### ISSUE TITLE
[bug] Error when running "conan test" with a lockfile

### ISSUE THREAD

**Issue by aleksa2808** [conan-io/conan#9088]
### Environment Details (include every applicable attribute)

Used the `conanio/gcc9:1.37.1` image for reproducibility.

### Steps to reproduce (Include if Applicable)

```bash
docker run -it --rm conanio/gcc9:1.37.1

conan profile new default --detect
conan profile update settings.compiler.libcxx=libstdc++11 default

git clone https://github.com/conan-io/conan-center-index
cd conan-center-index/recipes/abseil/all/ # first package usable on linux, got the same issue with others

conan lock create --ref abseil/20210324.2
conan test -l conan.lock test_package abseil/20210324.2
```

### Logs (Executed commands with output) (Include/Attach if Applicable)

```bash
conan@de15da3a5366:~/conan-center-index/recipes/abseil/all$ conan test -l conan.lock test_package abseil/20210324.2
Using lockfile: '/home/conan/conan-center-index/recipes/abseil/all/conan.lock'
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=gcc
compiler.libcxx=libstdc++11
compiler.version=9
os=Linux
os_build=Linux
[options]
[build_requires]
[env]

Traceback (most recent call last):
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/command.py", line 2176, in run
    method(args[0][1:])
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/command.py", line 308, in test
    lockfile=args.lockfile, profile_build=profile_build)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/conan_api.py", line 93, in wrapper
    return f(api, *args, **kwargs)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/conan_api.py", line 332, in test
    test_build_folder=test_build_folder, recorder=recorder)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/cmd/test.py", line 39, in install_build_and_test
    recorder=recorder)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/manager.py", line 60, in deps_install
    is_build_require=is_build_require)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/graph/graph_manager.py", line 120, in load_graph
    apply_build_requires=apply_build_requires)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/graph/graph_manager.py", line 272, in _resolve_graph
    graph_lock=graph_lock)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/graph/graph_manager.py", line 380, in _load_graph
    apply_build_requires=apply_build_requires)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/client/graph/graph_manager.py", line 365, in _recurse_build_requires
    new_profile_build_requires)
  File "/opt/pyenv/versions/3.7.5/lib/python3.7/site-packages/conans/model/graph_lock.py", line 568, in check_locked_build_requires
    locked_requires = locked_node.build_requires
AttributeError: 'NoneType' object has no attribute 'build_requires'

ERROR: 'NoneType' object has no attribute 'build_requires'
```

**Comment by memsharded** [comment]
Thanks for the useful report. Managed to reproduce and trying to provide a fix for it in https://github.com/conan-io/conan/pull/9089 (aiming for patch Conan 1.37.2, probably next week)

**Comment by czoido** [comment]
Closed by: https://github.com/conan-io/conan/pull/9089 to be released in 1.37.2 this week
